src/js/component.js

```diff
@@ -1425,16 +1425,20 @@ class Component {
    * Registers a component
    *
    * @param {String} name Name of the component to register
-   * @param {Object} comp The component to register
+   * @param {Object} Comp The component to register
    * @static
    * @method registerComponent
    */
-  static registerComponent(name, comp) {
-    if (!name) {
-      return;
+  static registerComponent(name, Comp) {
+    if (typeof name !== 'string' || !(/\S/).test(name)) {
+      throw new Error('illegal "name"; must be non-empty string');
     }
 
-    name = toTitleCase(name);
+    if (!Component.prototype.isPrototypeOf(Comp.prototype) && Component !== Comp) {
+      throw new Error('illegal "Comp"; must be a subclass of Component');
+    }
+
+    name = toTitleCase(name.trim());
```

Comment:
it might be worth warning that the component name contains trailing/leading spaces and that we are trimming it here?